lib/galaxy/datatypes/binary.py

```diff
@@ -423,6 +432,8 @@ def set_meta( self, dataset, overwrite=True, **kwd ):
             dataset.metadata.read_groups = [ read_group['ID'] for read_group in dataset.metadata.bam_header.get( 'RG', [] ) if 'ID' in read_group ]
             dataset.metadata.sort_order = dataset.metadata.bam_header.get( 'HD', {} ).get( 'SO', None )
             dataset.metadata.bam_version = dataset.metadata.bam_header.get( 'HD', {} ).get( 'VN', None )
+            dataset.metadata.columns = 12
+            dataset.metadata.column_types = ['str', 'int', 'str', 'int', 'int', 'str', 'str', 'int', 'int', 'str', 'str', 'str']
```

Comment:
Could not understand the importance of above two lines. Column names are displayed even if i removing these two lines.